extensions/positron-assistant/src/tools/projectTreeTool.ts

```diff
@@ -0,0 +1,287 @@
+ */
+function convertUrisToDirectoryItems(folder: vscode.WorkspaceFolder, uris: vscode.Uri[]): DirectoryItem {
+	// If there are no URIs, return an empty directory structure for the folder
+	if (uris.length === 0) {
+		return [folder.name, []];
+	}
+
+	// Sort the URIs alphabetically (folders and files are sorted together)
+	uris.sort((a, b) => a.fsPath.localeCompare(b.fsPath));
+
+	const root: Record<string, any> = {};
+	for (const uri of uris) {
+		// Get the path relative to the workspace folder, e.g. src/myfolder/myfile.txt
+		const relativePath = vscode.workspace.asRelativePath(uri, false);
+
+		// Split the relative path into segments, e.g. ['src', 'myfolder', 'myfile.txt']
+		const segments = relativePath.split('/');
+		let node = root;
+
+		// Iterate through the segments to build the directory structure
+		for (let i = 0; i < segments.length; i++) {
+			const segment = segments[i];
+			const isLastSegment = i === segments.length - 1;
+			if (isLastSegment) {
+				node[segment] = null;
+			} else {
+				node[segment] = node[segment] || {};
+				node = node[segment];
+			}
+		}
+	}
+
+	// Convert the nested object structure to an array of DirectoryItems
+	const toDirectoryItems = (node: Record<string, any>): DirectoryItem[] =>
+		Object.entries(node).map(([name, value]) =>
+			value === null ? name : [name, toDirectoryItems(value)]
+		);
+
+	return [folder.name, toDirectoryItems(root)];
+}
+
+/**
+ * Convert from the tool input excludeSettings to the vscode.ExcludeSettingOptions
+ * @param excludeSetting The exclude setting from the tool input.
+ * @returns The corresponding vscode.ExcludeSettingOptions value.
+ */
+function getExcludeSettingOptions(excludeSetting: string) {
+	switch (excludeSetting) {
+		case '':
+			return vscode.ExcludeSettingOptions.None;
+		case 'filesExclude':
+			return vscode.ExcludeSettingOptions.FilesExclude;
+		case 'searchAndFilesExclude':
+			return vscode.ExcludeSettingOptions.SearchAndFilesExclude;
+		default:
+			return DEFAULT_EXCLUDE_SETTING_OPTIONS;
+	}
+}
+
+// Default values for the project tree tool options
+const DEFAULT_MAX_FILES = 500;
+const DEFAULT_FILTER_RESULTS = true;
+const DEFAULT_USE_IGNORE_FILES = { local: true, parent: true, global: true };
+const DEFAULT_EXCLUDE_SETTING_OPTIONS = vscode.ExcludeSettingOptions.SearchAndFilesExclude;
+const DEFAULT_INCLUDE_PATTERNS = ['**/*'];
+const DEFAULT_EXCLUDE_PATTERNS = [
+	// Directories
+	'**/.build/**',
+	'**/.git/**',
+	'**/.devcontainer/**',
+	'**/.hg/**',
+	'**/.ipynb_checkpoints/**',
+	'**/.pytest_cache/**',
+	'**/.svn/**',
+	'**/.venv/**',
+	'**/.Rproj.user/**',
+	'**/.vscode/**',
+	'**/__pycache__/**',
+	'**/dist/**',
+	'**/node_modules/**',
+	'**/renv/**',
+	'**/venv/**',
+	// Files
+	'.DS_Store',
+	'.RData',
+	'.Rhistory',
+	'desktop.ini',
+	'package-lock.json',
+	'yarn.lock',
+	'Thumbs.db',
+	// Wildcards
+	'**/*.a',
+	'**/*.bak',
+	'**/*.csv~',
+	'**/*.js.map',
+	'**/*.log',
+	'**/*.o',
+	'**/*.pyo',
+	'**/*.so',
+	'**/*.tmp',
+];
```

Comment:
any other default exclude patterns to add here?

should this be moved into a config setting?

Reply:
I wonder if we should be excluding some of these files e.g. `yarn.lock`, `*.csv`, `*.log`. They could be useful to answer questions about the project like "is this an npm package" or "is X a dependency of this project".

I would lean to keeping this hardcoded for now and waiting for some user feedback.